### ISSUE TITLE
Crash when samba connection drops

### ISSUE THREAD

**Issue by thomastaylor312** [Microsoft/vscode#515]
On a Mac (running OSX 10.11.1), if the network connection to a connected Samba share with open files interrupts, VSCode crashes with the error that says it needs to restart.

**Comment by bpasero** [comment]
@thomastaylor312 does it work with Atom?

**Comment by thomastaylor312** [comment]
@bpasero I am new to VSCode, so just double checking, you mean Github Atom, correct?

**Comment by thomastaylor312** [comment]
@bpasero Assuming it is Github Atom, it doesn't crash when I replicate the same situation. I am trying to get the logs for VSCode, but I don't know where they are located.

**Comment by bpasero** [comment]
@thomastaylor312 yes Atom editor. Maybe run Code.exe from the command line from its program folder with "--verbose" to capture the output.

**Comment by thomastaylor312** [comment]
Here is the verbose output
```
./Electron --verbose
### VSCode main.js ###
/Applications/Visual Studio Code.app/Contents/Resources/app { pathArguments: [],
  programStart: 0,
  workers: -1,
  enablePerformance: false,
  verboseLogging: true,
  debugPluginHostPort: undefined,
  debugBrkPluginHost: undefined,
  logPluginHostCommunication: false,
  firstrun: false,
  openNewWindow: false,
  openInSameWindow: false,
  gotoLineMode: false,
  pluginHomePath: undefined,
  pluginDevelopmentPath: undefined,
  pluginTestsPath: undefined }
Lifecycle#unload() 1
IPC#vscode-workbenchLoaded
IPC#vscode-openFileFolderPicker
IPC#vscode-openFileFolderPicker
App#activate
App#activate
App#activate
IPC#vscode-openFileFolderPicker
Lifecycle#unload() 1
IPC#vscode-workbenchLoaded
IPC#vscode-openNewWindow
Lifecycle#unload() 2
IPC#vscode-workbenchLoaded
Lifecycle#window-before-close 2
Lifecycle#unload() 2
Lifecycle#window-before-close 2
Lifecycle#window-close 2
IPC#vscode-openNewWindow
Lifecycle#unload() 3
IPC#vscode-workbenchLoaded
IPC#vscode-openFileFolderPicker
[VS Code]: render process crashed!
App#activate
Lifecycle#before-quit
Lifecycle#window-before-close 1
Lifecycle#unload() 1
Lifecycle#window-before-close 1
Lifecycle#window-close 1
Lifecycle#window-all-closed
Lifecycle#before-quit
App#dispose: deleting running instance handle
```

**Comment by bpasero** [comment]
Can someone send me instructions how to setup such an environment? I would like to reproduce.

**Comment by thomastaylor312** [comment]
1. On a Mac running OSX 10.11.1 (probably a MacBook Pro) and the latest VSCode, connect to a Samba share in the Finder.
2. Open a file from the Samba share for editing
3. Sleep the computer or disconnect from an Ethernet connection. Normally I have seen it happen when I close the lid on my machine, leave it closed for a few minutes (not connected to power), and and then open it back up
4. When you wake up the machine the dialog box pops up saying that VSCode had an issue and needs to close.

Note: I have replicated this various times with my own machine.

**Comment by thomastaylor312** [comment]
@bpasero I was just checking back on the status of this. Also, I attached a partial error report when it crashed again.
[vscode_crash.txt](https://github.com/Microsoft/vscode/files/67179/vscode_crash.txt)

**Comment by thomastaylor312** [comment]
Just checking back again on the status of this. Do you need any other help from me?

**Comment by bpasero** [comment]
I suggest to report this to either Electron or node.js as these components are crashing. This would require however to have reproducible steps in a bare Electron app.

**Comment by bpasero** [comment]
@thomastaylor312 looking at the attached crash report, I wonder if the crash report is actually related to the samba issue. Could it be a crash when you are in fullscreen mode and then restore from sleep?

**Comment by thomastaylor312** [comment]
@bpasero It has happened in and out of full screen mode. I have tried to dig around for more failure messages, but I haven't been able to find any

**Comment by andersonferrari** [comment]
I am also having the same problema with opening files on a virtual machine on my computer through SAMBA... it would be nice to have a fix... maybe find a way to configure Samba not to drop the connection if that's the case... 
thanks!

**Comment by bpasero** [comment]
What setup is this, what virtual machine? Parallels?

**Comment by andersonferrari** [comment]
My host OS is Windows 10 with a Ubuntu Desktop 12.04 running on a hyper-v box. 
On the guest OS (linux) I have my application servers running, and my sorce code. Sorce code is exposed (shared) with host through SAMBA (I work with GitHub on the guest OS and I enter my linux/samba user when I first make the connection from Host). 
It all works fine most of the time, except when I leave Visual Studio code alone for a few minutes it breaks with that REOPEN message. From other posts I have read, I'm guessing it may be some problem with losing connection momentarily with SAMBA.

**Comment by bpasero** [comment]
@andersonferrari would you be able to help me out and setup vs code to run out of the sourcecode as described at https://github.com/Microsoft/vscode/wiki/How-to-Contribute so that we can see where the error is coming from?

**Comment by bpasero** [comment]
Once you can run scripts\code.bat from the command line, open the workspace with the samba share and wait for it to crash, you should get the crash dump as output to the console.

**Comment by bpasero** [comment]
@thomastaylor312 @andersonferrari  when this crash happens you can use the Mac Console application to find the crash log:

<img width="1414" alt="screen shot 2016-03-24 at 08 04 18" src="https://cloud.githubusercontent.com/assets/900690/14010617/08c57ce0-f197-11e5-9c79-16aef29b1bac.png">

**Comment by thomastaylor312** [comment]
@bpasero I haven't been on samba in a while editing stuff. Let me try again and then I'll upload the log I find.

**Comment by bpasero** [comment]
Thanks

**Comment by thomastaylor312** [comment]
@bpasero I have been unable to duplicate this again. However, I did update OS X since we first discovered the problem, so maybe it was a bug fix with connected shares? Grasping at straws here.

**Comment by bpasero** [comment]
@thomastaylor312 that would be very nice, please try how it behaves for a while!

**Comment by thomastaylor312** [comment]
I'll keep trying for the next week and let you know if I see anything

**Comment by thomastaylor312** [comment]
@bpasero I have still not seen any issues. @andersonferrari are you still seeing the problem?

**Comment by andersonferrari** [comment]
Hi Taylor,Thank you for your interest.I have upgraded to version 0.10.11... after that the error is not happening so often anymore... every 3 or 4 days that I keep VS Code opened I will have a crash, but it is probably some issue with the Samba in the VM (hyper-v) itself.A few days ago I had the issue and I went to the Windows Event Viewer (I am on Win10) but I didn't find anything specific to that error (maybe I was looking in the wrong place).thanks,Anderson

Date: Mon, 4 Apr 2016 10:19:27 -0700

**Comment by bpasero** [comment]
I suggest we close this and reopen when it happens again. Note though that there was no actual code change to fix the samba issue, it might still be there.
